test/unit/features/options/extends.spec.js

```diff
@@ -46,4 +46,41 @@ describe('Options extends', () => {
     expect(vm.b).toBe(2)
     expect(vm.c).toBe(3)
   })
+
+  it('should work with global mixins and Object.prototype.watch polyfill', done => {
+    if (!Object.prototype.watch) {
+      // eslint-disable-next-line no-extend-native
+      Object.prototype.watch = { remove: true }
+    }
+
+    Vue.use({
+      install: function () {
+        Vue.mixin({})
+      }
+    })
+    const spy = jasmine.createSpy('watch')
+    const A = Vue.extend({
+      data: function () {
+        return { a: 1 }
+      },
+      watch: {
+        a: spy
+      },
+      created: function () {
+        this.a = 2
+      }
+    })
+    new Vue({
+      extends: A
+    })
+    waitForUpdate(() => {
+      delete Object.prototype.watch
+      expect(spy).toHaveBeenCalledWith(2, 1)
+    }).then(done)
+  })
+  afterEach(function () {
```

Comment:
I think we could group this test into a describe and use `afterAll` so the check is run only once. You can also setup the `watch` property on the `beforeAll`